src/app/modules/group/components/group-manager-list/group-manager-list.component.ts

```diff
@@ -46,17 +54,14 @@ export class GroupManagerListComponent implements OnChanges, OnDestroy {
     private confirmationService: ConfirmationService,
   ) {}
 
-  ngOnChanges(_changes: SimpleChanges): void {
+  ngOnChanges(changes: SimpleChanges): void {
     if (this.groupData) {
-      this.groupId$.next(this.groupData.group.id);
+      (changes.groupData?.previousValue as GroupData | undefined)?.group.id !== this.groupData?.group.id
+        ? this.reloadData()
+        : this.fetchData();
```

Comment:
This is probably overkill because I don't have any case in mind for which the `groupId` will change **and** the component won't be destroyed+remounted. However it can be considered future-safe.